### PR TITLE
Refactor to use templates more

### PR DESCRIPTION
**Pull request by oprypin**
- Revert skipping broken themes
- refactor: Consistently run mkdocs from CLI
- refactor: Avoid unnecessary plugins
- refactor: devnull is part of standard library
- refactor: More widely apply Jinja templates for specimen

### CODE REVIEW

**Comment by oprypin** [comment]
* Blocked on https://github.com/pawamoy/mkdocs-gallery/issues/8

File: build.py [hunk 1]
```diff
@@ -176,19 +105,19 @@ def install_deps(theme: Theme) -> None:
...
-            .split("\n")
-        )
-        subprocess.run([venv_dir / "bin" / "pip", "install", *deps], check=False, stdout=devnull, stderr=devnull)
+    deps = subprocess.run(
+        [sys.executable, "-mmkdocs", "get-deps", "-f", theme_dir / "mkdocs.yml"],
+        check=False,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.DEVNULL,
+        text=True,
+    ).stdout.splitlines()
```
Comment (pawamoy) [inline build.py:114]: I remember having errors because of the trailing new line returning an empty dependency `""`, hence the `strip` call.

**Review by pawamoy — APPROVED** [review]
Super nice, thanks a lot!

File: build.py (hunk 1 above)
Comment (oprypin) [reply build.py:114]: `splitlines` deals with the last newline instead now

Comment (pawamoy) [reply build.py:114]: OK, awesome :rocket:

**Comment by oprypin** [comment]
(would prefer this to be merged without squashing)

**Pull request merged**